### PR TITLE
sql: add twin-driver and support postgres normalize

### PR DESCRIPTION
**Pull request by a8m**

### CODE REVIEW

File: sql/internal/sqlx/twin.go [hunk 1]
```diff
@@ -2,25 +2,40 @@
...
+	// DropClause holds optional clauses that
+	// can be added to the DropSchema change.
+	DropClause []schema.Clause
+}
+
+// NormalizeRealm implements the schema.Normalizer interface.
+//
+// The implementation converts schema objects in "natural form" (e.g. HCL or DSL)
+// to their "normal presentation" in the database, by creating them temporarily in
+// a "twin database", and then inspect then from there.
```
Comment (hedwigz) [inline sql/internal/sqlx/twin.go:37]:
```suggestion
// a "twin database", and then inspects them from there.
```

File: sql/internal/sqlx/twin.go [hunk 2]
```diff
@@ -56,46 +71,44 @@ func (d *Driver) NormalizeRealm(ctx context.Context, r *schema.Realm) (nr *schem
...
-
-func twinName(name string) string {
+func (t *TwinDriver) formatName(name string) string {
 	twin := fmt.Sprintf("atlas_twin_%s_%d", name, time.Now().Unix())
-	if len(twin) <= maxLen {
+	if t.MaxNameLen == 0 || len(twin) <= t.MaxNameLen {
 		return twin
 	}
-	return fmt.Sprintf("%s_%x", twin[:maxLen-33], md5.Sum([]byte(twin)))
+	return fmt.Sprintf("%s_%x", twin[:t.MaxNameLen-33], md5.Sum([]byte(twin)))
```
Comment (hedwigz) [inline sql/internal/sqlx/twin.go:113]: maybe introduce some const `MD5_HEX_LEN = 32`

Comment (a8m) [reply sql/internal/sqlx/twin.go]: `md5.Size * 2` as `%x` uses two chars per byte.

File: sql/internal/sqlx/twin.go [hunk 3]
```diff
@@ -2,25 +2,40 @@
...
+	// DropClause holds optional clauses that
+	// can be added to the DropSchema change.
+	DropClause []schema.Clause
+}
+
+// NormalizeRealm implements the schema.Normalizer interface.
+//
+// The implementation converts schema objects in "natural form" (e.g. HCL or DSL)
+// to their "normal presentation" in the database, by creating them temporarily in
+// a "twin database", and then inspects then from there.
```
Comment (hedwigz) [inline sql/internal/sqlx/twin.go:37]:
```suggestion
// a "twin database", and then inspects them from there.
```

File: sql/internal/sqlx/twin.go (hunk 2 above)
Comment (hedwigz) [reply sql/internal/sqlx/twin.go]: HEX == two chars per byte

**Review by hedwigz — APPROVED** [review]

**Pull request merged**